torch/compiler/config.py

```diff
@@ -63,5 +63,13 @@
 A common use case for such a tag is to break caches.
 """
 
+dynamic_sources: str = Config(
```

Comment:
can we clearly define and document the precedence of test over dynamic = False for example and over config.assume_static_by_default ..
and over assume params static and assume nn module attributes static and add tests to guarantee the ordering

Reply:
Yeah let me do that in a separate PR that I'll stack on top of this one.